src/dmd/cond.d

```diff
@@ -362,13 +374,49 @@ extern (C++) final class StaticForeach : RootObject
         auto catass = new CatAssignExp(aloc, new IdentifierExp(aloc, idres), res[1]);
         s2.push(createForeach(aloc, pparams[1], new ExpStatement(aloc, catass)));
         s2.push(new ReturnStatement(aloc, new IdentifierExp(aloc, idres)));
-        auto aggr = wrapAndCall(aloc, new CompoundStatement(aloc, s2));
-        sc = sc.startCTFE();
-        aggr = aggr.expressionSemantic(sc);
-        aggr = resolveProperties(sc, aggr);
-        sc = sc.endCTFE();
-        aggr = aggr.optimize(WANTvalue);
-        aggr = aggr.ctfeInterpret();
+
+        Expression aggr;
+        Type indexty;
```

Comment:
```suggestion
        Expression aggr = void;
        Type indexty = void;
```

Reply:
Really?
What's the recommended usage of that.

Reply:
`aggr` is guaranteed to have a value, so can skip default initialization.

Reply:
Well yes, but it isn't some big struct just a pointer. I would expect the same binary output on GDC/LDC.

Reply:
I don't think there's any given rule that must be followed, just apply where it makes most sense.

Reply:
> Well yes, but it isn't some big struct just a pointer. I would expect the same binary output on GDC/LDC.

It still produces an extra instruction to set the value to null.  Don't expect the optimizer to elide it. :-)